src/VisualStudio/Core/Def/Implementation/Remote/ServiceHubRemoteHostClient.WorkspaceHost.cs

```diff
@@ -84,13 +70,44 @@ public void OnSolutionRemoved()
                 UnregisterPrimarySolution(solutionId, synchronousShutdown: false);
             }
 
+            private void RegisterPrimarySolution()
+            {
+                this.AssertIsForeground();
+                _currentSolutionId = _workspace.CurrentSolution.Id;
+                var solutionId = _currentSolutionId;
+
+                var storageLocation = _workspace.DeferredState?.ProjectTracker.GetWorkingFolderPath(_workspace.CurrentSolution);
+
+                try
+                {
+                    // this will only return false if OOP doesn't exist. otherwise,
+                    // it will throw and handled as usual
+                    _session.TryInvokeAsync(
+                        nameof(IRemoteHostService.RegisterPrimarySolutionId),
+                        new object[] { solutionId, storageLocation }, CancellationToken.None).Wait(CancellationToken.None);
+                }
+                catch (RemoteHostClientExtensions.UnexpectedRemoteHostException)
```

Comment:
Ditto: expected that this catch would have moved into TryInvokeAsync.

Reply:
same as above.